### PR TITLE
[PP-2] Create basic application structure

### PR DESCRIPTION
**Pull request by MakFill**
**Task**: https://andrei-lysiuk.atlassian.net/browse/PP-2
**Description**: created basic application structure by create-react-app typescript template. 
  Added:
  - `.gitignore`
  - `src` path alias
  - `tsconfig.json`
   - env files

### CODE REVIEW

File: tsconfig.paths.json [hunk 1]
```diff
@@ -0,0 +1,11 @@
+{
+  "compilerOptions": {
+    "baseUrl": ".",
+    "paths": {
+      "@components/*": ["./src/components/*"],
+      "@assets": ["./src/assets/*"],
+      "@pages/*": ["./src/pages/*"],
+      "@services": ["./src/services/*"]
```
Comment (mrakofff) [inline tsconfig.paths.json]: Leave only 1 alias: `"@/*"`: [`"src/*"]`

File: src/reportWebVitals.ts [hunk 2]
```diff
@@ -0,0 +1,15 @@
+import { ReportHandler } from 'web-vitals';
```
Comment (mrakofff) [inline src/reportWebVitals.ts]: Do we need this file?

File: package.json [hunk 3]
```diff
@@ -0,0 +1,47 @@
+{
+  "name": "pointing-poker-fe",
+  "version": "0.1.0",
+  "private": true,
+  "dependencies": {
+    "@testing-library/jest-dom": "^5.11.4",
+    "@testing-library/react": "^11.1.0",
+    "@testing-library/user-event": "^12.1.10",
+    "@types/jest": "^26.0.15",
```
Comment (mrakofff) [inline package.json]: move all `@types/*` dependencies to `devDependencies`

File: .gitignore [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+/build
+
+# misc
+.DS_Store
+.env.local
+.env.development.local
+.env.test.local
+.env.production.local
+
+npm-debug.log*
```
Comment (mrakofff) [inline .gitignore]: Could be replaced with `*.log` pattern

File: .gitignore [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+# dependencies
+/node_modules
+/.pnp
+.pnp.js
+
+# testing
+/coverage
+
+# production
+/build
```
Comment (mrakofff) [inline .gitignore:12]: Do we have `build` folder as production folder? Plz add `dist` folder too.

File: .gitignore [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+
+# misc
+.DS_Store
```
Comment (mrakofff) [inline .gitignore:16]: Plz add `.idea` pattern to exclude `WebStorm` files

**Review by mrakofff — CHANGES_REQUESTED** [review]
Add some comments

File: tsconfig.paths.json (hunk 1 above)
Comment (MakFill) [reply tsconfig.paths.json]: -  "@components/*": ["./src/components/*"],
    -  "@assets": ["./src/assets/*"],
    -  "@pages/*": ["./src/pages/*"],
    -  "@services": ["./src/services/*"]
    +  "@/*": ["./src/*"]
#

File: src/reportWebVitals.ts (hunk 2 above)
Comment (MakFill) [reply src/reportWebVitals.ts]: reportWebVitals.ts deleted

File: package.json (hunk 3 above)
Comment (MakFill) [reply package.json]: all @types/* dependencies placed into devDependencies

File: .gitignore (hunk 4 above)
Comment (MakFill) [reply .gitignore]: *.log instead:
npm-debug.log*
yarn-debug.log*
yarn-error.log*

File: .gitignore (hunk 5 above)
Comment (MakFill) [reply .gitignore:12]: added /dist folder

File: .gitignore (hunk 6 above)
Comment (MakFill) [reply .gitignore:16]: added .idea pattern

File: src/index.tsx [hunk 7]
```diff
@@ -0,0 +1,13 @@
+import React from 'react';
+import ReactDOM from 'react-dom';
+import App from './App';
+import reportWebVitals from './reportWebVitals';
```
Comment (mrakofff) [inline src/index.tsx]: Remove this import and usage if file was deleted

File: package.json [hunk 8]
```diff
@@ -0,0 +1,47 @@
...
+    "typescript": "^4.1.2",
+    "web-vitals": "^1.0.1"
+  },
+  "scripts": {
+    "start": "react-app-rewired start",
+    "build": "react-app-rewired build",
+    "test": "react-app-rewired test",
+    "eject": "react-app-rewired eject"
+  },
+  "eslintConfig": {
```
Comment (mrakofff) [inline package.json]: move `eslint` config to the separate file

File: package.json [hunk 9]
```diff
@@ -0,0 +1,47 @@
...
+    "extends": [
+      "react-app",
+      "react-app/jest"
+    ]
+  },
+  "browserslist": {
```
Comment (mrakofff) [inline package.json]: move `browserslist` to separate file

**Review by mrakofff — CHANGES_REQUESTED** [review]
Some little refactoring updates:)

File: package.json (hunk 9 above)
Comment (MakFill) [reply package.json]: move browserslist to .browserslistrc

File: package.json (hunk 8 above)
Comment (MakFill) [reply package.json]: add .eslintrc.json remove eslintConfig from packaje.json

File: src/index.tsx (hunk 7 above)
Comment (MakFill) [reply src/index.tsx]: Completed

**Review by mrakofff — APPROVED** [review]

**Review by Praddby — APPROVED** [review]

**Pull request merged**